### DISCUSSION TITLE
How to scan trivy repo with specific branch

### DISCUSSION THREAD

**Discussion by ontakaspi** [aquasecurity/trivy#2270] [Q&A] [answered]
Hi,
I'm currently interested in trivy scan git repository, the scanning works in standard format URL repo, but my problem is, how to scan with the specific branch on the SCM repo, in GitLab I've tried this but nothing worked except standard scan format URL which is default branch.

![image](https://user-images.githubusercontent.com/49944982/172142654-a8655d1b-de25-49cd-9348-52de5163e4d7.png)

*Note: I've set up gitlab_token as it's a private repo.

Please let me know about this.
Thanks!

**Comment by eyk107** [comment]
any news on this topic?

**Comment by fatihtokus** [comment]
Hi

How do you specify the repo and branch while starting the scan?

**Comment by knqyf263** [comment, marked as answer]
Please refer to the doc.
https://aquasecurity.github.io/trivy/v0.46/docs/target/repository/#scanning-a-branch
